addons/product/tests/test_variants.py

```diff
@@ -622,17 +623,24 @@ def test_variant_images(self):
         """Check that on variant, the image used is the image_variant_1920 if set,
         and defaults to the template image otherwise.
         """
+        # pretend setup happened in an older transaction
+        self.variants.write({
+            'write_date': self.cr.now() - timedelta(milliseconds=1),
+        })
```

Comment:
I tried running the test on my computer and this doesn't actually work.

`write_date`/`__last_update` value of variants is the same before and after this instruction for me, and the touch doesn't update the value either during the test.

I did rebase on latest master but I'm not sure if this had an impact.

Reply:
I actually got the same issue when running it locally this time.
After investigating, I found out it actually depends on how you tag the test to make it run: `write` ignores the updates of log access fields like `write_date` if the pool is already initialized. By not tagging the test with the usual `@odoo.tests.common.tagged('post_install', '-at_install')` it works, because it then runs `at_install`.

If we really want to make this work `post_install`, then we will have to mock `cr.now()` during the setup or during the test.

Reply:
I see, thanks for the feedback. It would be better if it worked in all situations of course.
It should be fine to change the `now` in the test, that wouldn't be the first one to do it.

Reply:
> write ignores the updates of log access fields like write_date if the pool is already initialized.

Doesn't it work with `_write()` as I suggested in private message? I am pretty sure I have this multiple times, and IMHO better than patching the `now()`.
If that doesn't work ok..

Reply:
I now understood why it did not work when I tried with `_write()`: it's a direct SQL operation.
I had to invalidate the cache for the patched `write_date` to be taken into account.

Reply:
Interesting solution, it would be worth adding a short comment explaining why using `_write` instead of `write` above the line.

And maybe check if you can give fnames and ids to `invalid_cache` to avoid a complete clean? Because otherwise `flush` should be called before, to avoid losing all other pending updates on all other models/fields.